### PR TITLE
fix: flickering nfc button

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- fixes flickering nfc button on receive screen

### CODE REVIEW

File: app/screens/receive-bitcoin-screen/receive-screen.tsx [hunk 1]
```diff
@@ -37,36 +37,30 @@ const ReceiveScreen = () => {
...
     })()
-  }, [
-    LL.ReceiveScreen,
-    colors.black,
-    navigation,
-    request?.state,
-    request?.type,
-    styles.nfcIcon,
-  ])
+  }, [nfcText, colors.black, navigation, request?.state, request?.type])
```
Comment (nicolasburtey) [inline app/screens/receive-bitcoin-screen/receive-screen.tsx:63]: warning @UncleSamtoshi

**Comment by nicolasburtey** [comment]
what was the issue here? how are we fixing it?

**Comment by UncleSamtoshi** [comment]
The issue was that both `LL.ReceiveScreen` and `styles.nfcIcon` were new objects on every render, so it was causing the useEffect to trigger over and over again. During those triggers I think there were some race conditions where it was causing the button to flicker. It was fixed by relying directly and on the translations string (which doesnt change) and removing `styles.nfcIcon` from the useEffect dependency array which should have no effect because the styles should not change.

File: app/screens/receive-bitcoin-screen/receive-screen.tsx (hunk 1 above)
Comment (UncleSamtoshi) [reply app/screens/receive-bitcoin-screen/receive-screen.tsx:65]: Disabled the warning.

**Review by nicolasburtey — APPROVED** [review]

**Comment by nicolasburtey** [comment]
this might be where using `why-did-you-render` would solve those issues or get us warning

**Pull request merged**